### PR TITLE
Add incremental build support to crossgen-corelib

### PR DESCRIPTION
**Pull request by agocke**

### CODE REVIEW

File: src/coreclr/crossgen-corelib.proj [hunk 1]
```diff
@@ -2,37 +2,40 @@
...
+  <PropertyGroup Condition="'$(BuildArchitecture)' != '$(TargetArchitecture)'">
+    <CrossDir Condition="'$(TargetArchitecture)' == 'arm' or '$(TargetArchitecture)' == 'arm64'">x64</CrossDir>
+    <CrossDir Condition="'$(TargetArchitecture)' == 'arm' and '$(UseCrossgen2)' != 'true' and '$(OS)' == 'Windows_NT'">x86</CrossDir>
+    <CrossDir Condition="'$(TargetArchitecture)' == 'x86' and '$(UseCrossgen2)' == 'true'">$(BuildArchitecture)</CrossDir>
+  </PropertyGroup>
+  <PropertyGroup>
+    <CrossGen1ToolPath>$([MSBuild]::NormalizePath('$(BinDir)', '$(CrossDir)', 'crossgen$(ExeExtension)'))</CrossGen1ToolPath>
+    <CrossGen2ToolPath>$([MSBuild]::NormalizePath('$(BinDir)', '$(CrossDir)', 'crossgen2', 'crossgen2.dll'))</CrossGen2ToolPath>
+    <CrossGenToolPath Condition="'$(UseCrossgen2)' == 'false'">$(CrossGen1ToolPath)</CrossGenToolPath>
+    <CrossGenToolPath Condition="'$(UseCrossgen2)' != 'false'">$(CrossGen2ToolPath)</CrossGenToolPath>
```
Comment (jkoritzinsky) [inline src/coreclr/crossgen-corelib.proj:32]: We should make sure to include some of the ILCompiler.* assemblies in the up-to-date check. That's where most of the crossgen2 logic lives.

Comment (agocke) [reply src/coreclr/crossgen-corelib.proj:32]: Do you use auto-generated reference assemblies during build? Unless you do, the exe should always be rebuilt if the dependencies change

Comment (jkoritzinsky) [reply src/coreclr/crossgen-corelib.proj:32]: To be honest I’m not sure if we do in this part of the tree.

Comment (agocke) [reply src/coreclr/crossgen-corelib.proj:32]: Confirmed -- this works fine. Ref assembly building is not enabled. In this case, I'd say enabling is a mistake. The primary benefit is not rebuilding the downstream tree, but in this case we almost always want to rebuild the downstream tree.

**Comment by agocke** [comment]
Sven beat me to it in https://github.com/dotnet/runtime/pull/48086

**Pull request closed without merging**